api/types/src/account.rs

```diff
@@ -25,3 +28,43 @@ impl From<AccountResource> for AccountData {
         }
     }
 }
+
+/// An Enum for referencing an asset type, either coin or fungible asset.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum AssetType {
+    Coin(MoveStructTag),
+    FungibleAsset(Address),
+}
+
+impl FromStr for AssetType {
+    type Err = anyhow::Error;
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match Address::from_str(s) {
+            Ok(address) => Ok(AssetType::FungibleAsset(address)),
+            Err(_) => {
+                match MoveStructTag::from_str(s) {
+                    Ok(tag) => Ok(AssetType::Coin(tag)),
+                    Err(e) => Err(e),
+                }
+            }
+        }
+    }
+}
```

Comment:
Oh, I see, you do one then the other

Reply:
yes, I do FA first as it is shorter and we will migrate mostly eventually.